src/js/redirection.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @name Redirection
+ * @desc A Redirection object can be instanciated with an array of strings as first parameter or
+ * @desc with languages as parameters (spread argument)
+ * @desc First language will be selected as default language
+ */
+class Redirection { // eslint-disable-line no-unused-vars
+  /**
+   * @constructor
+   * @example
+   * new Redirection([ 'fr-FR', 'en-US' ])
+   * or
+   * new Redirection('fr-FR', 'en-US')
+   */
+  constructor () {
+    this.pages = arguments.length === 1 && Array.isArray(arguments[0]) ? arguments[0] : Array.from(arguments)
+    if (!this.pages.length) {
+      throw new Error('RedirectionError: There should be at least one language in the constructor call')
+    }
+    if (!this.pages.every(lang => /^([a-z]{2}-[A-Z]{2})$/.test(lang))) {
+      throw new Error('RedirectionError: All languages should have the following format: aa-AA with aa, the language and AA, the region')
+    }
+    this.lang2page = {}
+    this.pages.forEach(page => {
+      const lang = page.split('-')[0]
+      if (!this.lang2page[lang]) {
+        this.lang2page[lang] = page
+      }
+    })
+  }
+
+  /**
+   * @method go
+   * @param  {String} [url]      Optional sub URL - real URL will be 'folderLanguage/url'
+   * @param  {Number} [cooldown] Optional timer cooldown before redirection (in seconds)
+   * @example
+   * redirection.go('index.html') // Redirecting to /lang/index.html
+   * or
+   * redirection.go('index.html', 2) // Redirecting to /lang/index.html after 2 seconds
+   */
+  go (url, cooldown) {
+    const userlangs = window.navigator.languages || window.navigator.language || window.navigator.userLanguage
+    for (const lang of Array.isArray(userlangs) ? userlangs : [ userlangs ]) {
```

Comment:
Consider using consistent styling? `[ userlangs ]` for a array, and line 44 `[lang]` for a array too ?

Reply:
`[lang]` is not an array; it is a property accessor...